introduction_to_amazon_algorithms/imageclassification_caltech/Image-classification-transfer-learning-highlevel.ipynb

```diff
@@ -54,8 +54,11 @@
     "print(role)\n",
```

Comment:
<p>For the sake of clarity, I would split prediction and result parsing. Something like in the example below.</p><p> </p><p>Also, the list of classes is placed in the middle of the block of code that computes and prints the result. It could be moved somewhere else to not look like a distraction. Maybe in a dedicated cell right above this one?</p><p> </p><p>I like how the notebook is clean and detailed. It doesn't do more than necessary to show the idea, and at the same time is quite verbose about important details, like hyperparameters. As an optional improvement, comments can be added to the code to explain what is going on. Since there will be users with different background, it may be helpful for some of them. Example below illustrates my idea, it's not necessary to follow it exactly.</p><p> </p><pre class="ql-syntax" spellcheck="false"># Invoke the deployed model to compute prediction
prediction = ic_classifier.predict(payload, initial_args = {'ContentType':'application/x-image'})

# "prediction" is a JSON string. Load it into a Python object.
# The result will contain a list of probabilities for all classes.
probabilities = json.loads(prediction)

# Find the class with the maximum probability
predicted_category_index = np.argmax(probabilities)

# Print the category name and confidence score
predicted_category_name = object_categories[index]
confidence = result[predicted_category_index]
print(f"Result: label - {predicted_category_name}, probability - {confidence}")
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1700/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
<p>Above looks like a good idea. Incorporated everything except the f-string. Will change it once there is confirmation about python/python3.</p>